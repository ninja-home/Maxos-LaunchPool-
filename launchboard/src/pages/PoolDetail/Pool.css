.blank {
    width: 100vw;
    background-color: white;
}

.pool-detail-body {
    width: 100vw;
    min-height: 100vh;
    background-color: white;
}

.loading {
    width: 300px !important;
    margin-top: 100px !important;
    margin-bottom: 100px !important;
    margin-left: calc(50vw - 150px) !important;
}

.loading1 {
    width: 300px !important;
    margin-top: calc(20vh - 100px) !important;
    margin-bottom: 100px !important;
    margin-left: calc(50vw - 182px) !important;
}

.loading2 {
    width: 300px !important;
    margin-top: calc(20vh - 100px) !important;
    margin-bottom: 100px !important;
    margin-left: calc(50vw - 144px) !important;
}

::-webkit-scrollbar {
    width: 20px;
}
::-webkit-scrollbar-track {
    background-color: transparent;
}
::-webkit-scrollbar-thumb {
    background-color: #d6dee1;
    border-radius: 20px;
    border: 6px solid transparent;
    background-clip: content-box;
}
::-webkit-scrollbar-thumb:hover {
    background-color: #a8bbbf;
}

@media only screen and (min-width: 1440px) {  
    .pool-detail-body {
        background-color: white;
    }
    .risk-free {
        margin-left: 48px !important;
    }
    .stake-list-body {
        margin: 0 32px 30px 32px !important;
        overflow-y: auto;
        height: 40vh;
    }
    .connect-dialog-title {
      padding: 16px 32px 8px 32px !important;
      width: 890px !important;
    }
    .loading1 {
        width: 300px !important;
        margin-top: 80px !important;
        margin-bottom: 100px !important;
    }
    .pool-divider {
        margin: 30px !important;
        width: calc(100% - 60px) !important;
    }
}

.stake-list-title {
    margin: 30px 42px 30px 42px !important;
}

.stake-list-body {
    margin: 0 32px 30px 32px !important;
    overflow-y: auto;
    height: 40vh;
    
    border: solid;
    border-width: thin;
    border-color: #DDDCDC;
    border-radius: 10px;
}

.pool-divider {
    margin: 4vw;
    width: calc(100% - 8vw);
}

.risk-free {
    margin-left: 4vw;
    font-family: sans-serif;
    font-weight: 600;
    font-style: normal;
    font-size: 16px;
    line-height: 20px;
    color: #4BC9FF;
}

.connect-dialog {
    margin: 0px;
    bottom: 0;
    position: absolute;
}

.connect-dialog-title {
    font-family: sans-serif;
    font-style: normal;
    font-weight: 600;
    font-size: 24px;
    line-height: 30px;
    cursor: move;
    width: calc(100vw - 70px);
    color: #0377A8;
}

.connect-dialog-close {
    cursor: pointer;
    float: right;
    margin-top: 5px;
    width: 20px;
}