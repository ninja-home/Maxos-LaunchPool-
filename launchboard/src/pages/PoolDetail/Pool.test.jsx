import { render } from '@testing-library/react';
import Pool from './index';
import { Provider } from "react-redux";
import store from "../../store";

import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('PoolDetailPage tests', () => {
    it('PoolDetailPage renders without craching', () => {
        const pooldetailpage = render(<Provider store={store}><Pool /></Provider>);
        expect(pooldetailpage).toMatchSnapshot();
    });
    it('Connect Wallet button click shows dialog', () => {
        Enzyme.configure({ adapter: new Adapter() });
        const pooldetailpage = shallow(<Provider store={store}><Pool/></Provider>);
        setTimeout(async() => {
            const button = pooldetailpage.find('#connect-wallet-button');
            button.simulate('click');
            expect(pooldetailpage.find('#connect-wallet-modal-dialog').prop('isOpen')).toBe(true);
        }, 1000);  
    });
    it('Connect Metamask button click shows correct info', () => {
        Enzyme.configure({ adapter: new Adapter() });
        const pooldetailpage = shallow(<Provider store={store}><Pool/></Provider>);
        setTimeout(async() => {
            const button = pooldetailpage.find('#metamask-connect-button');
            button.simulate('click');
            expect(pooldetailpage.find('#connect-wallet-modal-dialog').prop('isOpen')).toBe(false);
            expect(pooldetailpqt3.find('#stake-form-connect-wallet-name').textContent).toEqual('Metamask Wallet');
        }, 1000);  
    });
    it('Connect Coinbase button click shows correct info', () => {
        Enzyme.configure({ adapter: new Adapter() });
        const pooldetailpage = shallow(<Provider store={store}><Pool/></Provider>);
        setTimeout(async() => {
            const button = pooldetailpage.find('#coinbase-connect-button');
            button.simulate('click');
            expect(pooldetailpage.find('#connect-wallet-modal-dialog').prop('isOpen')).toBe(false);
            expect(pooldetailpqt3.find('#stake-form-connect-wallet-name').textContent).toEqual('Coinbase Wallet');
        }, 1000);  
    });
    it('Connect Another button click shows correct info', () => {
        Enzyme.configure({ adapter: new Adapter() });
        const pooldetailpage = shallow(<Provider store={store}><Pool/></Provider>);
        setTimeout(async() => {
            const button = pooldetailpage.find('#another-connect-button');
            button.simulate('click');
            expect(pooldetailpage.find('#connect-wallet-modal-dialog').prop('isOpen')).toBe(false);
            expect(pooldetailpqt3.find('#stake-form-connect-wallet-name').textContent).toEqual('Another Wallet');
        }, 1000);  
    });
    it('Email validation is working correctly in the Stake Form Component', () => {
        Enzyme.configure({ adapter: new Adapter() });
        const pooldetailpage = shallow(<Provider store={store}><Pool/></Provider>);
        setTimeout(async() => {
            const email = pooldetailpage.find('#email');
            email.simulate('change', { target: { value: 'a' } });
            expect(pooldetailpage.find('#connect-wallet-butto').prop('disabled')).toBe(true);
            email.simulate('change', { target: { value: 'a@a' } });
            expect(pooldetailpage.find('#connect-wallet-butto').prop('disabled')).toBe(true);
            email.simulate('change', { target: { value: 'a.a' } });
            expect(pooldetailpage.find('#connect-wallet-butto').prop('disabled')).toBe(true);
            email.simulate('change', { target: { value: 'a@a.' } });
            expect(pooldetailpage.find('#connect-wallet-butto').prop('disabled')).toBe(true);
            email.simulate('change', { target: { value: '@a.a' } });
            expect(pooldetailpage.find('#connect-wallet-butto').prop('disabled')).toBe(true);
            email.simulate('change', { target: { value: 'a@a.a' } });
            expect(pooldetailpage.find('#connect-wallet-butto').prop('disabled')).toBe(false);
        }, 1000);  
    })
    it('Amount validation is working correctly in the Stake Form Component', () => {
        Enzyme.configure({ adapter: new Adapter() });
        const pooldetailpage = shallow(<Provider store={store}><Pool/></Provider>);
        setTimeout(async() => {
            const amount = pooldetailpage.find('#amount');
            amount.simulate('change', { target: { value: 'a' } });
            expect(pooldetailpage.find('#connect-wallet-butto').prop('disabled')).toBe(true);
            amount.simulate('change', { target: { value: 'a1' } });
            expect(pooldetailpage.find('#connect-wallet-butto').prop('disabled')).toBe(true);
            amount.simulate('change', { target: { value: '1a' } });
            expect(pooldetailpage.find('#connect-wallet-butto').prop('disabled')).toBe(true);
            amount.simulate('change', { target: { value: '1.a' } });
            expect(pooldetailpage.find('#connect-wallet-butto').prop('disabled')).toBe(true);
            amount.simulate('change', { target: { value: '12' } });
            expect(pooldetailpage.find('#connect-wallet-butto').prop('disabled')).toBe(false);
            amount.simulate('change', { target: { value: '12.3' } });
            expect(pooldetailpage.find('#connect-wallet-butto').prop('disabled')).toBe(false);
            amount.simulate('change', { target: { value: '.3' } });
            expect(pooldetailpage.find('#connect-wallet-butto').prop('disabled')).toBe(false);
        }, 1000);  
    })
})

