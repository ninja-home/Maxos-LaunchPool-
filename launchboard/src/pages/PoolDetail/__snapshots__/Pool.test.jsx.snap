// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoolDetailPage tests PoolDetailPage renders without craching 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="blank"
      >
        <div
          class="pool-detail-body"
        >
          <div>
            <a>
              <img
                class="land-logo"
                src="land-logo.png"
              />
            </a>
          </div>
          <div
            class="otoco-body"
          >
            <img
              class="otoco-logo"
              src="another-pool.png"
            />
            <div>
              <div
                class="otoco-title"
              />
              <div
                class="otoco-content"
              >
                "Compliant corporate structures ‚Äúin-a-box‚Äù for DeFi"
              </div>
            </div>
          </div>
          <div
            class="pool-divider"
          >
            <hr
              class="MuiDivider-root"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
            >
              <div
                class="share-list-otoco"
              >
                <img
                  class="share-logo"
                  src="com-logo.png"
                />
                <a
                  class="share-description"
                >
                  www.otoco.com
                </a>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
            >
              <div
                class="share-list-discord"
              >
                <img
                  class="share-logo"
                  src="discord-logo.png"
                />
                <a
                  class="share-description"
                >
                  Chat on Discord
                </a>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
            >
              <div
                class="share-list-vote"
              >
                <img
                  class="share-logo"
                  src="vote-logo.png"
                  style="width: 10px; margin-left: 5vw; margin-right: 28px;"
                />
                <a
                  class="share-description"
                >
                  Vote
                </a>
              </div>
            </div>
          </div>
          <div
            class="pool-divider"
          >
            <hr
              class="MuiDivider-root"
            />
          </div>
          <div
            class="pool-detail-progress"
          >
            <div
              class="stake-info-body"
            >
              <div
                class="stake-info-list"
              >
                <div
                  class="stake-info-amount"
                  id="stake-info-stakes"
                />
                <div
                  class="stake-info-state"
                >
                  STAKES
                </div>
              </div>
              <div
                class="stake-info-list"
              >
                <div
                  class="stake-info-amount"
                  id="stake-info-staked"
                />
                <div
                  class="stake-info-state"
                >
                  STAKED
                </div>
              </div>
              <div
                class="stake-info-list"
              >
                <div
                  class="stake-info-amount"
                  id="stake-info-daysleft"
                />
                <div
                  class="stake-info-state"
                >
                  DAYS LEFT
                </div>
              </div>
            </div>
            <div
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="NaN"
              class="MuiLinearProgress-root WithStyles(ForwardRef(LinearProgress))-root-1 MuiLinearProgress-colorPrimary WithStyles(ForwardRef(LinearProgress))-colorPrimary-2 MuiLinearProgress-determinate"
              id="stake-info-progress"
              role="progressbar"
            >
              <div
                class="MuiLinearProgress-bar WithStyles(ForwardRef(LinearProgress))-bar-3 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
                style="transform: translateX(NaN%);"
              />
            </div>
            <div
              class="progress-info-body"
            >
              <div
                class="progress-info-min"
                id="stake-info-min"
              >
                 MIN
              </div>
              <div
                class="progress-info-max"
                id="stake-info-max"
              >
                 MAX
              </div>
            </div>
          </div>
          <div
            class="pool-divider"
          >
            <hr
              class="MuiDivider-root"
            />
          </div>
          <div
            class="risk-free"
          >
            üìñ How risk-free staking works
          </div>
          <div
            class="connect-wallet-body"
          >
            <div
              class="connect-wallet-info-body"
            >
              <div
                class="connect-wallet-info-top"
              >
                <div
                  class="connect-wallet-info-position"
                >
                  <div
                    class="connect-wallet-info-positive-state"
                  >
                    POSITION
                  </div>
                  <div
                    class="connect-wallet-info-positive-amount"
                    id="connect-wallet-info-position"
                  >
                    33
                  </div>
                </div>
                <div
                  class="connect-wallet-blank"
                />
                <div
                  class="connect-wallet-info-stake"
                >
                  <div
                    class="connect-wallet-info-stake-state"
                  >
                    STAKE
                  </div>
                  <div
                    class="connect-wallet-info-stake-amount"
                    id="connect-wallet-info-staked"
                  />
                </div>
              </div>
              <div
                class="connect-wallet-button"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text WithStyles(ForwardRef(Button))-root-4"
                  id="connect-wallet-button"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    CONNECT WALLET
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
          <h2
            class="stake-list-title"
          >
            All Stakes
          </h2>
          <div
            class="stake-list-body"
          >
            <div>
              <img
                class="loading1"
                src="loading.gif"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="blank"
    >
      <div
        class="pool-detail-body"
      >
        <div>
          <a>
            <img
              class="land-logo"
              src="land-logo.png"
            />
          </a>
        </div>
        <div
          class="otoco-body"
        >
          <img
            class="otoco-logo"
            src="another-pool.png"
          />
          <div>
            <div
              class="otoco-title"
            />
            <div
              class="otoco-content"
            >
              "Compliant corporate structures ‚Äúin-a-box‚Äù for DeFi"
            </div>
          </div>
        </div>
        <div
          class="pool-divider"
        >
          <hr
            class="MuiDivider-root"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
          >
            <div
              class="share-list-otoco"
            >
              <img
                class="share-logo"
                src="com-logo.png"
              />
              <a
                class="share-description"
              >
                www.otoco.com
              </a>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
          >
            <div
              class="share-list-discord"
            >
              <img
                class="share-logo"
                src="discord-logo.png"
              />
              <a
                class="share-description"
              >
                Chat on Discord
              </a>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
          >
            <div
              class="share-list-vote"
            >
              <img
                class="share-logo"
                src="vote-logo.png"
                style="width: 10px; margin-left: 5vw; margin-right: 28px;"
              />
              <a
                class="share-description"
              >
                Vote
              </a>
            </div>
          </div>
        </div>
        <div
          class="pool-divider"
        >
          <hr
            class="MuiDivider-root"
          />
        </div>
        <div
          class="pool-detail-progress"
        >
          <div
            class="stake-info-body"
          >
            <div
              class="stake-info-list"
            >
              <div
                class="stake-info-amount"
                id="stake-info-stakes"
              />
              <div
                class="stake-info-state"
              >
                STAKES
              </div>
            </div>
            <div
              class="stake-info-list"
            >
              <div
                class="stake-info-amount"
                id="stake-info-staked"
              />
              <div
                class="stake-info-state"
              >
                STAKED
              </div>
            </div>
            <div
              class="stake-info-list"
            >
              <div
                class="stake-info-amount"
                id="stake-info-daysleft"
              />
              <div
                class="stake-info-state"
              >
                DAYS LEFT
              </div>
            </div>
          </div>
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="NaN"
            class="MuiLinearProgress-root WithStyles(ForwardRef(LinearProgress))-root-1 MuiLinearProgress-colorPrimary WithStyles(ForwardRef(LinearProgress))-colorPrimary-2 MuiLinearProgress-determinate"
            id="stake-info-progress"
            role="progressbar"
          >
            <div
              class="MuiLinearProgress-bar WithStyles(ForwardRef(LinearProgress))-bar-3 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
              style="transform: translateX(NaN%);"
            />
          </div>
          <div
            class="progress-info-body"
          >
            <div
              class="progress-info-min"
              id="stake-info-min"
            >
               MIN
            </div>
            <div
              class="progress-info-max"
              id="stake-info-max"
            >
               MAX
            </div>
          </div>
        </div>
        <div
          class="pool-divider"
        >
          <hr
            class="MuiDivider-root"
          />
        </div>
        <div
          class="risk-free"
        >
          üìñ How risk-free staking works
        </div>
        <div
          class="connect-wallet-body"
        >
          <div
            class="connect-wallet-info-body"
          >
            <div
              class="connect-wallet-info-top"
            >
              <div
                class="connect-wallet-info-position"
              >
                <div
                  class="connect-wallet-info-positive-state"
                >
                  POSITION
                </div>
                <div
                  class="connect-wallet-info-positive-amount"
                  id="connect-wallet-info-position"
                >
                  33
                </div>
              </div>
              <div
                class="connect-wallet-blank"
              />
              <div
                class="connect-wallet-info-stake"
              >
                <div
                  class="connect-wallet-info-stake-state"
                >
                  STAKE
                </div>
                <div
                  class="connect-wallet-info-stake-amount"
                  id="connect-wallet-info-staked"
                />
              </div>
            </div>
            <div
              class="connect-wallet-button"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text WithStyles(ForwardRef(Button))-root-4"
                id="connect-wallet-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  CONNECT WALLET
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <h2
          class="stake-list-title"
        >
          All Stakes
        </h2>
        <div
          class="stake-list-body"
        >
          <div>
            <img
              class="loading1"
              src="loading.gif"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
